### DISCUSSION TITLE
BOLEClassifier

### DISCUSSION THREAD

**Discussion by danielnowakassis** [online-ml/river#1124] [Polls]
Hi. I've made an implementation of the Boosting Online Learning Ensemble (BOLE) (https://ieeexplore.ieee.org/document/7727427) in River (passed all tests). Should I do a pull request?

**Comment by MaxHalford** [comment]
Hey there 👋

In general, it's a good idea to contribute. Without seeing the code, the questions I usually have are:

- How performant is it? The paper doesn't have a lot of citations, so we need some benchmarks of some sort to make sure it really works
- Is it very different than all the existing methods in River? Does it necessarily have to be a new class? Or can it can be seen as a variation of an existing class?
- Is the code rather complex? Will we have a hard-time maintaining it?

**Reply by danielnowakassis** [reply]
Hey 👋 

1. There is an implementation of BOLE in the MOA framework by the authors. And my implementation is in my river fork(boosting.py). I just need to do some adjustments in predict_proba_one. I can do some experiments with the well-known data sets of the stream literature. 
2. No. But in my opinion, some methods are better to override.
3. The code is not complex.

By the way, Merry Christmas 🙂

**Reply by MaxHalford** [reply]
Sounds good :)

Feel free to open a pull request when you feel like it. We can go over the details. Generally speaking it would be nice to have a model that explicitly copes with drift. It would be a novel addition to River.
